operator-common/src/main/java/io/strimzi/operator/common/operator/resource/DeploymentConfigOperator.java

```diff
@@ -46,4 +46,35 @@ protected Integer currentScale(String namespace, String name) {
         desired.getSpec().getTemplate().getSpec().getContainers().get(0).setImage(current.getSpec().getTemplate().getSpec().getContainers().get(0).getImage());
         return super.internalPatch(namespace, name, current, desired);
     }
+
+    /**
+     * Asynchronously polls the deployment configuration until either the observed generation matches the desired
+     * generation sequence number or timeout.
+     *
+     * @param namespace The namespace.
+     * @param name The resource name.
+     * @param pollIntervalMs The polling interval
+     * @param timeoutMs The timeout
+     * @return  A future which completes when the observed generation of the deployment configuration matches the
+     * generation sequence number of the desired state.
+     */
+    public Future<Void> waitForObserved(String namespace, String name, long pollIntervalMs, long timeoutMs) {
```

Comment:
Should this logic be part of `readiness`? I.e. that _current version_ of the resource is ready?

Reply:
Yes, it should be, at least for Deployment objects. I should have thought of incorporating this into the readiness check when I wrote the method

Reply:
Since `waitForObserved` is already used as part of the KafkaAssemblyOperator, should we hold off on this change for a refactor?

Reply:
On second thought, the refactor can probably be done here, the PR is not as big as I originally thought